tests/test_geoserver.py

```diff
@@ -0,0 +1,219 @@
+import tempfile
+
+import numpy as np
+import pytest
+
+from ravenpy.utilities.testdata import get_local_testdata
+
+
+class TestHydroBASINS:
+    geoserver = pytest.importorskip("ravenpy.utilities.geoserver")
+
+    fiona = pytest.importorskip("fiona")
+    gpd = pytest.importorskip("geopandas")
+    sgeo = pytest.importorskip("shapely.geometry")
+
+    def test_select_hybas_na_domain(self):
+        bbox = (-68.0, 50.0) * 2
+        dom = self.geoserver.select_hybas_domain(bbox)
+        assert dom == "na"
+
+    def test_select_hybas_ar_domain(self):
+        bbox = (-114.65, 61.35) * 2
+        dom = self.geoserver.select_hybas_domain(bbox)
+        assert dom == "ar"
+
+    def test_get_hydrobasins_location_wfs(self):
+        lake_winnipeg = (-98.03575958286369, 52.88238524279493)
+        feature = self.geoserver.get_hydrobasins_location_wfs(
+            coordinates=lake_winnipeg * 2, lakes=True, domain="na"
+        )
+
+        gml = tempfile.NamedTemporaryFile(suffix=".gml", delete=False)
+        with open(gml.name, "wb") as f:
+            f.write(feature)
+
+        with self.fiona.open(gml.name) as src:
+            feat = next(iter(src))
+            geom = self.sgeo.shape(feat["geometry"])
+            assert geom.bounds == (-99.2731, 50.3603, -96.2578, 53.8705)
+            np.testing.assert_almost_equal(geom.area, 3.2530867)
+
+    def test_get_hydrobasins_attributes_wfs(self):
+        rio_grande = (-80.475, 8.4)
+        feature = self.geoserver.get_hydrobasins_location_wfs(
+            coordinates=rio_grande * 2, lakes=True, domain="na"
+        )
+
+        gml = tempfile.NamedTemporaryFile(suffix=".gml", delete=False)
```

Comment:
can't we use pytest tmp_path for stuff like that?

Reply:
Indeed, we can! I always forget about that function.

Reply:
I always forget too, but it's great, less verbose actually!